server/sonar-server/src/main/java/org/sonar/server/authentication/ws/LogoutAction.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.authentication.ws;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.sonar.api.web.ServletFilter;
+import org.sonar.server.authentication.JwtHttpHandler;
+import org.sonar.server.authentication.event.AuthenticationEvent;
+import org.sonar.server.authentication.event.AuthenticationException;
+
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+
+public class LogoutAction extends ServletFilter {
+
+  public static final String AUTH_LOGOUT_URL = "/api/authentication/logout";
+
+  private static final String POST = "POST";
+
+  private final JwtHttpHandler jwtHttpHandler;
+  private final AuthenticationEvent authenticationEvent;
+
+  public LogoutAction(JwtHttpHandler jwtHttpHandler, AuthenticationEvent authenticationEvent) {
+    this.jwtHttpHandler = jwtHttpHandler;
+    this.authenticationEvent = authenticationEvent;
+  }
+
+  @Override
+  public UrlPattern doGetPattern() {
+    return UrlPattern.create(AUTH_LOGOUT_URL);
+  }
+
+  @Override
+  public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain) throws IOException, ServletException {
+    HttpServletRequest request = (HttpServletRequest) servletRequest;
+    HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+    if (!request.getMethod().equals(POST)) {
+      response.setStatus(HTTP_BAD_REQUEST);
+      return;
+    }
+    generateAuthenticationEvent(request, response);
+    jwtHttpHandler.removeToken(request, response);
+  }
+
+  private void generateAuthenticationEvent(HttpServletRequest request, HttpServletResponse response) {
+    try {
+      Optional<JwtHttpHandler.Token> token = jwtHttpHandler.getToken(request, response);
+      String userLogin = token.isPresent() ? token.get().getUserDto().getLogin() : null;
+      authenticationEvent.logout(request, userLogin, null);
+    } catch (AuthenticationException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)